### DISCUSSION TITLE
Option to display border and background around sections

### DISCUSSION THREAD

**Discussion by BBE-FR** [home-assistant/frontend#22288] [Dashboard] [answered]
Section are a very powerful way to create dashboard.

It could become even more versatile with some customizing possibilities, like the capability to define, at section level, if a border or background should be displayed around the whole section itself.

Either with the "simple" option, in each section, to have have 2 selectors to display or hide both:
- a border 
- a background
Or even a color pickup option to define in the interface the border/ background color.

At section level, this could allow even very basic user to visually group cards within a section.
This would be fully compatible with the grid design, and would preserve the capability to use or not this feature depending the cases.

Currently the "grouping" effect of section is only visually ensured by enlarged spacing between section. Background / Border could make it more obvious.  

Section edit could be enriched with these options.
![image](https://github.com/user-attachments/assets/7d1810bc-8a58-4fd4-bf0b-a0fa63dc483f)

**Comment by BBE-FR** [comment]
This would allow even basic user to group cards in a very simple way, and using header cards and buttons create very efficients sort of "new card", without the need to some custom vertical-stack-in-cards or stack-in-cards:
![image](https://github.com/user-attachments/assets/151ff095-a8a4-436e-b0d6-f48d923311b8)

**Comment by awilsonhass** [comment]
I do this with card-mod, but extremely agree this should be native. It’s much too difficult as it stands. 
![Screenshot 2024-12-03 at 8 35 51 AM jpeg](https://github.com/user-attachments/assets/9f5bbaf6-b71c-4783-972b-9b0c73547071)

**Reply by ndom91** [reply]
Can you share more? I'm having trouble targetting the grid section itself with card-mod :thinking:

**Reply by BBE-FR** [reply]
The code was shared here: https://community.home-assistant.io/t/wth-i-cant-specify-a-background-color-per-section/803363

It is using a custom mod card, not modding the section itself.

I do personally almost the same with a vertical-stack-in-card, that does not require full YAML for user less confident with YAML.

But the aim of this FR for me is to get rid of custom components to be able to perform it...

**Reply by ndom91** [reply]
Great, thanks a lot

**Comment by nagtegaal** [comment]
Wouldn't introducing the `style:` attribute to the yaml be better and more flexible? 
IMO this should be available in all cards, sections, pages and so on... This would make card-mod redundant and makes things more flexible, consistent and easier to personalize/theme.

**Comment by ghost** [comment]
I fully agree that there should be an easy way to define border and/or background for sections. 
This would increase the clarity of the dashboard.

**Comment by bucky2780** [comment]
I really wish borders and backgrounds were supported.
+1

**Comment by ZorK766** [comment]
+1

**Comment by marcinbauer-ohf** [comment]
Hi!
This is an interesting discussion and we will explore adding this as a feature for sections. Stay tuned in the next week for initial designs on how this might work.

**Reply by BBE-FR** [reply]
Promising teaser !!

**Reply by pcan08** [reply]
Full support for this FR, after the teaser, a trailer ?

**Reply by BBE-FR** [reply]
We stayed tuned... but I have seen nothing in 2025.3 and 2025.4... Did I miss something ?

**Reply by silamon** [reply]
@marcinbauer85

**Comment by officedrone** [comment]
Looking forward to this one! Sections are already great and adding the ability to border and custom-background them would take them to the next level!

**Comment by butterford** [comment]
Was just trying to achieve exactly this in new section dashboard! Please implement.

**Comment by alexsaas** [comment]
THAT(!) would be really perfect. I would love it.

**Comment by NPo1973** [comment]
Yes please for this feature - just created a section that I want in RED as a warning to avoid clicking on this unless absolutely necessary..

(basically powers of the plug running the network switch - kind of an emergency kill switch.  Need it on a specific dashboard but don't want to accidentally click it while not thinking.

**Reply by Misiu** [reply]
You can add confirmation to those cards (https://www.home-assistant.io/dashboards/actions/#options-for-confirmation)
```
tap_action:
  action: toggle
  confirmation:
    text: Are you Sure?
```
Ideal for things that you don't want to turn off accidentally.

**Comment by TidjyDev** [comment]
I don't know why this is something HA developers haven't implemented since. Everyone knows it's easy to implement for someone who knows a lot about code and has no impact on the dashboard's functionality. So why are we still waiting for this feature to be available? I mean, every end users want and need this!

**Reply by BBE-FR** [reply]
It is not that easy if you really think about it.

If you want to draw a background around a specific section and not all section, you have to slightly change the layout to add some additional spacing (padding).

exemple in the horizontal padding, but same is true in vertical:
<img width="806" height="685" alt="image" src="https://github.com/user-attachments/assets/ff87d4fc-b5f8-4b98-8d18-cce3f036c5c9" />

If you don't do this padding, you need more space between sections for everybody which would be lost space for most of the user and increased complexity for all, or if you dont implement any padding for the background, it will be ugly, with either sections separated, but cards "touching" the border of the sections, or sections not separated one from another...

Bref, it is feasible, but not that simple to do it in the right way.

**Reply by vatioz** [reply]
I can see the complexity creeps up once you get into the details.
Wouldn't be an option to have two modes - borderless (current) and with borders (for all sections). It's kind of a middle ground and users could decide if they value space or clarity.

**Comment by BBE-FR** [comment]
Subject transferred there: https://github.com/orgs/home-assistant/discussions/492

**Reply by silamon** [reply]
There's also an open pull request right now, I'm not sure if for the entire feature request or just for a part, but things may move on this one.

**Reply by BBE-FR** [reply]
Exact https://github.com/home-assistant/frontend/pull/26369

Lets see where it goes...

**Reply by Nickduino** [reply]
> Exact #26369
> 
> Lets see where it goes...

It was merged into dev, then reverted, never came back

**Comment by alexsaas** [comment]
It's coming! And I love this change, thanks so much to @piitaya: https://github.com/home-assistant/frontend/pull/30228
It could make custom stack-in-cards obsolete.

However, I still see some areas that could be improved, but I'm not sure where to post them, so I'm posting them here:

1. There is no option to select the card background color as the background color for the whole section.

2. It should be possible to disable the borders, any box shadows set via the theme, and the background color of the embedded cards. In the screenshots, it looks as though the borders remain visible even with colored backgrounds. That might be fine, but sometimes it looks unattractive.

It would be great if these additions could be implemented.

**Comment by BBE-FR** [comment, marked as answer]
Implemented by @piitaya Thanks !!

**Comment by Nickduino** [comment]
It **is** great. It made other issues come to light, though:
https://github.com/orgs/home-assistant/discussions/3443
https://github.com/home-assistant/frontend/issues/51619
